sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -122,7 +122,12 @@ class StaxXmlParser(
       }
       val parser = StaxXmlParserUtils.filteredReader(xml)
       val rootAttributes = StaxXmlParserUtils.gatherRootAttributes(parser)
-      Some(convertObject(parser, schema, options, rootAttributes))
+      // A structure object is an attribute-only element
+      // if it only consists of attributes and valueTags.
+      val isRootAttributesOnly = schema.fields.forall { f =>
+        f.name == options.valueTag || f.name.startsWith(options.attributePrefix)
+      }
+      Some(convertObject(parser, schema, options, rootAttributes, isRootAttributesOnly))
```

Comment:
Couldn't we just pass `true` instead of `isRootAttributesOnly` here? There is a check for `nameToIndex.get(options.valueTag)` below anyway.

Reply:
We only read values if the struct only consists of attributes and `valueTag`. If there's another field, the value in between the tags will be ignored.